modules/tinymce/src/core/main/ts/commands/IndentOutdent.ts

```diff
@@ -49,6 +52,18 @@ const canOutdent = (editor: Editor): boolean => {
   return !editor.mode.isReadOnly() && (blocks.length > 1 || validateBlocks(editor, blocks));
 };
 
+const canIndent = (editor: Editor): boolean => {
+  const max = Options.getListMaxDepth(editor);
+  return !editor.mode.isReadOnly() && max.map((max) => {
+    const blocks = getSelectedListRoots(editor);
+    if (blocks.length > 0) {
+      return parseLists(SugarElements.fromDom(blocks), getItemSelection(editor)).some((entry) => entry.entries.some((value) => value.depth < max));
```

Comment:
I was about to chide you for using the native `Array.some` method instead of `Arr.exists`

But surely you don't need to parse the entire list and iterate through every node just to pull out the depth. These `can` methods are checked after every node change; that could get quite expensive.

Think about what depth represents and how you might be able to quickly figure it out using the selected element (then if necessary, apply it to multiple selected list items, perhaps similar to how `canOutdent` does).

Reply:
I was uncertain of how far you can rely on a `SelectorFilter.ancestors().length` and not wanting desyncs between the actual capability and this one as it could lead to false positives and false negatives in the case of some funky setup. And at that point I didn't want to basically duplicate the code of the parseLists to just find the depth.
But we could do it the swift way and handle any edgecases if they pop up if you think that's acceptable?